include/highfive/bits/H5Node_traits_misc.hpp

```diff
@@ -198,6 +201,55 @@ inline bool NodeTraits<Derivate>::exist(const std::string& group_path) const {
     return _exist(group_path);
 }
 
+
+static inline LinkType _convert_link_type(const H5L_type_t& ltype) {
+    switch (ltype) {
+        case H5L_TYPE_HARD:
+            return LinkType::Hard;
+        case H5L_TYPE_SOFT:
+            return LinkType::Hard;
+        case H5L_TYPE_EXTERNAL:
+            return LinkType::Hard;
+        default:
+            throw std::invalid_argument("Invalid Link type ");
```

Comment:
Error handling in this PR seems inconsistent: sometimes the `default` cases of the `switch` statements return _invalid value_, sometimes it throws.
It is my understanding that HighFive relies on exception for error handling, so should this PR.

Reply:
This is an internal function that eventually doesn't even make sense to throw.. we dont want to use exceptions for flow control, right? On the other hand I also don't like enums with the "Invalid" entry.
What's your suggestion @tristan0x ?

Reply:
If a function is not supposed to handle all cases, do not specify a `default` case, or put an `assert(true)` to be explicit about the lack of code.

Modern compilers will emit a warning if you forget to update a `switch(my_enum)` statement after adding a value to the enum.